substrate/frame/staking-async/src/tests/unbonding_queue.rs

```diff
@@ -0,0 +1,603 @@
+			for i in current_era + 1..bonding_duration {
+				assert_ok!(Staking::bond_extra(RuntimeOrigin::signed(11), 100));
+				assert_ok!(Staking::bond_extra(RuntimeOrigin::signed(21), 100));
+				assert_ok!(Staking::bond_extra(RuntimeOrigin::signed(31), 100));
+				assert_eq!(Staking::get_min_lowest_stake(), 500);
+				Session::roll_until_active_era(i + 1);
+				assert_eq!(Staking::get_min_lowest_stake(), 500);
+			}
+
+			// After this the lowest value will have been removed, and next iterations the
+			// number will increase by 100 each era.
+			let current_era = Staking::current_era();
+			for i in current_era + 1..bonding_duration {
+				assert_ok!(Staking::bond_extra(RuntimeOrigin::signed(11), 100));
+				assert_ok!(Staking::bond_extra(RuntimeOrigin::signed(21), 100));
+				assert_ok!(Staking::bond_extra(RuntimeOrigin::signed(31), 100));
+				assert_eq!(Staking::get_min_lowest_stake(), (500 + 100 * (i - 1)).into());
+				Session::roll_until_active_era(i + 1);
+				assert_eq!(Staking::get_min_lowest_stake(), (500 + 100 * i).into());
+			}
+		});
+}
+
+#[test]
+fn get_min_lowest_stake_with_many_validators_works() {
+	ExtBuilder::default()
+		.validator_count(9)
+		.multi_page_election_provider(3)
+		// We already have the following validators:
+		// 11 -> 1000
+		// 21 -> 1000
+		.set_status(31, StakerStatus::<AccountId>::Validator) // 1000
+		.set_status(41, StakerStatus::<AccountId>::Validator) // 1000
+		.set_status(51, StakerStatus::<AccountId>::Validator) // 1000
+		.add_staker(61, 100, StakerStatus::<AccountId>::Validator)
+		.add_staker(71, 200, StakerStatus::<AccountId>::Validator)
+		.add_staker(81, 1000, StakerStatus::<AccountId>::Validator)
+		.add_staker(91, 1000, StakerStatus::<AccountId>::Validator)
+		// Add nominators
+		.set_status(101, StakerStatus::<AccountId>::Nominator(vec![11])) // 500
+		.add_staker(102, 500, StakerStatus::<AccountId>::Nominator(vec![21]))
+		.add_staker(103, 500, StakerStatus::<AccountId>::Nominator(vec![31]))
+		.add_staker(104, 500, StakerStatus::<AccountId>::Nominator(vec![41]))
+		.add_staker(105, 500, StakerStatus::<AccountId>::Nominator(vec![51]))
+		.add_staker(106, 500, StakerStatus::<AccountId>::Nominator(vec![61]))
+		.add_staker(107, 500, StakerStatus::<AccountId>::Nominator(vec![71]))
+		.add_staker(108, 500, StakerStatus::<AccountId>::Nominator(vec![81]))
+		.add_staker(109, 500, StakerStatus::<AccountId>::Nominator(vec![91]))
+		.build_and_execute(|| {
+			assert_ok!(Staking::set_staking_configs(
+				RuntimeOrigin::root(),
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Set(UnbondingQueueConfig {
+					min_slashable_share: Perbill::from_percent(50),
+					lowest_ratio: Perbill::from_percent(34),
+					unbond_period_lower_bound: 1,
+					back_of_unbonding_queue: Zero::zero(),
+				})
+			));
+
+			assert_eq!(Staking::get_min_lowest_stake(), 0);
```

Comment:
Also I don't like naming things exactly as per the RFC, as they might not translate well in the code. 

For example, the name `get_min_lowest_stake` doesn't imply what this value exactly is, unless if the RFC is read.

Reply:
So... I agree with you: it's not intuitive enough. But the method represents the minimum stake in the last 28 eras that covers the lowest third validators with the lowest stake. How can you possible find a name to express all that and not to occupy a whole line?

I hoped it'd suffice with the method's description, which I'll happily change if needed be for a more descriptive version. If not... does any function name ring a bell?